tdrs-backend/tdpservice/parsers/duplicate_manager.py

```diff
@@ -0,0 +1,193 @@
+            return self.record_ids
+        return dict()
+
+    def __generate_error(self, err_msg, record, schema, has_precedence, is_new_max_precedence):
+        """Add an error to the managers error dictionary.
+
+        @param err_msg: string representation of the error message
+        @param record: a Django model representing a datafile record
+        @param schema: the schema from which the record was created
+        @param has_precedence: boolean indicating if this incoming error has equivalent precedence to current errors
+        @param is_new_max_precedence: boolean indicating if this error has the new max precedence
+        """
+        if has_precedence:
+            error = self.generate_error(
+                        error_category=ParserErrorCategoryChoices.CASE_CONSISTENCY,
+                        schema=schema,
+                        record=record,
+                        field=None,
+                        error_message=err_msg,
+                    )
+            if is_new_max_precedence:
+                self.manager_error_dict[self.my_hash] = [error]
+            else:
+                self.manager_error_dict.setdefault(self.my_hash, []).append(error)
+            self.num_errors = len(self.manager_error_dict[self.my_hash])
+
+    def __get_partial_hash(self, record):
+        partial_hash = None
+        if record.RecordType in {"T1", "T4"}:
+            partial_hash = hash(record.RecordType + str(record.RPT_MONTH_YEAR) + record.CASE_NUMBER)
+        elif record.RecordType in {"T2", "T3", "T5"}:
+            partial_hash = hash(record.RecordType + str(record.RPT_MONTH_YEAR) + record.CASE_NUMBER +
+                                str(record.FAMILY_AFFILIATION) + record.DATE_OF_BIRTH + record.SSN)
+        return partial_hash
+
+    def add_case_member(self, record, schema, line, line_number):
+        """Add case member and generate errors if needed.
+
+        @param record: a Django model representing a datafile record
+        @param schema: the schema from which the record was created
+        @param line: the raw string line representing the record
+        @param line_number: the line number the record was generated from in the datafile
+        @return: the number of duplicate errors
+        """
+        if self.current_line_number is None or self.current_line_number != line_number:
+            self.current_line_number = line_number
+            self.record_ids.setdefault(schema.document, []).append(record.id)
+            is_exact_dup = False
+            err_msg = None
+            has_precedence = False
+            is_new_max_precedence = False
+
+            line_hash = hash(line)
+            if line_hash in self.record_hashes:
+                has_precedence, is_new_max_precedence = self.error_precedence.has_precedence(ErrorLevel.DUPLICATE)
+                existing_record_line_number = self.record_hashes[line_hash]
+                err_msg = (f"Duplicate record detected with record type "
```

Comment:
@reitermb and @victoriaatraft could I get some UX input on the structure and content of the new error message?

Reply:
Exact duplicate